unittests/Swift/BuildSystemBindingsTests.swift

```diff
@@ -0,0 +1,158 @@
+//
+
+import XCTest
+
+// The Swift package has llbuildSwift as module
+#if SWIFT_PACKAGE
+import llbuild
+@testable import llbuildSwift
+#else
+@testable import llbuild
+#endif
+
+extension Command {
+  func with(name: String? = nil, shouldShowStatus: Bool? = nil, description: String? = nil, verboseDescription: String? = nil) -> Command {
+    return Command(name: name ?? self.name, shouldShowStatus: shouldShowStatus ?? self.shouldShowStatus, description: description ?? self.description, verboseDescription: verboseDescription ?? self.verboseDescription)
+  }
+}
+
+class BuildSystemBindingsTests: XCTestCase {
+  
+  func testCommand() {
+    let command = Command(name: "aName", shouldShowStatus: true, description: "a description", verboseDescription: "a verbose description")
+    XCTAssertEqual(command.name, "aName")
+    XCTAssertTrue(command.shouldShowStatus)
+    XCTAssertEqual(command.description, "a description")
+    XCTAssertEqual(command.verboseDescription, "a verbose description")
+    
+    XCTAssertEqual(command, command.with())
+    XCTAssertNotEqual(command, command.with(name: "foobar"))
+    XCTAssertNotEqual(command, command.with(shouldShowStatus: false))
+    XCTAssertNotEqual(command, command.with(description: "another description"))
+    XCTAssertNotEqual(command, command.with(verboseDescription: "another verbose description"))
+    
+    XCTAssertEqual(command.hashValue, command.with().hashValue)
+    XCTAssertNotEqual(command.hashValue, command.with(name: "foobar").hashValue)
+    XCTAssertNotEqual(command.hashValue, command.with(shouldShowStatus: false).hashValue)
+    XCTAssertNotEqual(command.hashValue, command.with(description: "another description").hashValue)
+    XCTAssertNotEqual(command.hashValue, command.with(verboseDescription: "another verbose description").hashValue)
+  }
+  
+  func testCommandMetrics() {
+    let metrics = CommandMetrics(utime: 1, stime: 2, maxRSS: 3)
+    XCTAssertEqual(metrics.utime, 1)
+    XCTAssertEqual(metrics.stime, 2)
+    XCTAssertEqual(metrics.maxRSS, 3)
+  }
+  
+  func testCommandExtendedResult() {
+    let metrics = CommandMetrics(utime: 1, stime: 2, maxRSS: 3)
+    
+    let result = CommandExtendedResult(result: .succeeded, exitStatus: 0, pid: 123, metrics: metrics)
+    XCTAssertEqual(result.result, .succeeded)
+    XCTAssertEqual(result.exitStatus, 0)
+    XCTAssertEqual(result.pid, 123)
+    XCTAssertEqual(result.metrics?.utime, 1)
+    XCTAssertEqual(result.metrics?.stime, 2)
+    XCTAssertEqual(result.metrics?.maxRSS, 3)
+    
+    var cExtendedResult = llb_buildsystem_command_extended_result_t(result: .succeeded, exit_status: 0, utime: 1, stime: 2, maxrss: 3, pid: 123)
+    let result2 = CommandExtendedResult(&cExtendedResult)
+    XCTAssertEqual(result2.result, .succeeded)
+    XCTAssertEqual(result2.exitStatus, 0)
+    XCTAssertEqual(result2.pid, 123)
+    XCTAssertEqual(result2.metrics?.utime, 1)
+    XCTAssertEqual(result2.metrics?.stime, 2)
```

Comment:
FWIW, I think this level of testing detail is overkill (checking that an unit function preroperly knitted)

Reply:
I agree, there's not really something to break here in the future and removing it also makes it possible to avoid the @testable (see above).